docs/examples/tdigest.c

```diff
@@ -0,0 +1,65 @@
+// cc -o tdigest tdigest.c $CARGO_TARGET_DIR/debug/libtimescaledb_toolkit_tdigest.a -lm -lpthread -ldl
+
+// Sample program which prints the expected output of the test_tdigest_io test.
+
+////////////////////////////////////////////////////////////////////////////////
+// TODO Generate a header from tdigest-c crate.
+
+#include <sys/types.h>
+
+struct TDigestBuilder;
+struct TDigest;
+
+// Return pointer to new TDigestBuilder.
+// MUST NOT be passed to free(3).  Instead, pass to timescaledb_toolkit_tdigest_builder_free to
+// discard or to timescaledb_toolkit_tdigest_build to convert to TDigest.
+// Never returns NULL.
+struct TDigestBuilder *
+timescaledb_toolkit_tdigest_builder_with_size(size_t size);
+
+void
+timescaledb_toolkit_tdigest_push(struct TDigestBuilder *builder, double value);
+
+void
+timescaledb_toolkit_tdigest_merge(struct TDigestBuilder *builder, struct TDigestBuilder *other);
+
+// Free a TDigestBuilder that has not been built.
+// MUST NOT be passed NULL.
+void
+timescaledb_toolkit_tdigest_builder_free(struct TDigestBuilder *builder);
+
+// Return pointer to new TDigest built from builder.
+// builder MUST NOT be passed to timescaledb_toolkit_tdigest_builder_free .
+struct TDigest *
+timescaledb_toolkit_tdigest_build(struct TDigestBuilder *builder);
+
+// Free a TDigest.
+void
+timescaledb_toolkit_tdigest_free(struct TDigest *td);
+
+// Return pointer to null-terminated buffer containing ASCII serialization of TDigest suitable for
+// use with postgresql INSERT.
+// Free the buffer with free(3).
+char *
+timescaledb_toolkit_tdigest_format_for_postgres(struct TDigest *td);
+
+////////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+
+int
+main()
+{
+    struct TDigestBuilder *builder = timescaledb_toolkit_tdigest_builder_with_size(100);
+    double value;
+    for (value = 1.0; value <= 100.0; value++) {
+        timescaledb_toolkit_tdigest_push(builder, value);
+    }
+
+    struct TDigest *td = timescaledb_toolkit_tdigest_build(builder);
+    printf("%s\n", timescaledb_toolkit_tdigest_format_for_postgres(td));
```

Comment:
We should free the return value from `timescaledb_toolkit_tdigest_format_for_postgres` in our example here to avoid leaking memory. It's not actually a problem here since the OS will clean it up after `main` anyways, but we don't want users to copy this example and forget to `free` this buffer.